plugins/inventory/gql_inventory.py

```diff
@@ -186,43 +186,60 @@ def add_variable(self, host: str, var: str, var_type: str):
+            device_attr = device.get(parent_attr)
+            if device_attr is None:
+                self.display.display(f"Could not find value for {parent_attr} on device {device_name}")
+                continue
+
+            if not chain:
+                group_name = device_attr
+
+            while chain:
+                group_name = chain.pop(0)
+                if isinstance(device_attr.get(group_name), Mapping):
+                    device_attr = device_attr.get(group_name)
                     continue
-                for attribute in group_by_path[1:]:
-                    if not isinstance(attr_value, Mapping):
-                        error_path = group_by.split(attribute)[0][:-1]
-                        self.display.display(f"Device {device_name} attribute {error_path} is not a dictionary.")
-                        continue
+                else:
                     try:
-                        attr_value = attr_value[attribute]
+                        group_name = device_attr[group_name]
                     except KeyError:
-                        self.display.display(f"Could not find value for {attribute} in {group_by} on device {device_name}")
+                        self.display.display(f"Could not find value for {group_name} in {group_by_path} on device {device_name}")
                         break
-                if isinstance(attr_value, Mapping):
-                    try:
-                        attr_value = attr_value["name"]
-                        self.display.display(f"No name value for {attr_value} on device {device_name} for group {group_by}.")
-                    except KeyError:
-                        try:
-                            attr_value = attr_value["slug"]
-                        except KeyError:
-                            self.display.display(f"No slug value for {attr_value} on device {device_name} for group {group_by}.")
-                            continue
-                if isinstance(attr_value, str):
-                    self.inventory.add_group(attr_value)
-                    self.inventory.add_child(attr_value, device_name)
+
+            if isinstance(group_name, Mapping):
+                if "name" in group_name:
+                    group_name = group_name["name"]
+                elif "slug" in group_name:
+                    group_name = group_name["slug"]
                 else:
-                    self.display.display(f"{attr_value} is not a valid group name for device {device_name} for group {group_by}.")
+                    self.display.display(f"No slug or name value for {group_name} in {group_by_path} on device {device_name}.")
+
+            if isinstance(group_name, str):
+                self.inventory.add_group(group_name)
+                self.inventory.add_child(group_name, device_name)
+            else:
+                self.display.display(f"{group_name} is not a valid group name for device {device_name}.")
```

Comment:
Wouldn't the error at 221-222 catch that? I thought this error was for if you whatever value it was trying group by wasn't a string.

Reply:
I think the code handles mappings and strings. If we know it won't be one of those two. It may be helpful to provide a clearer message. Something to the affect that it didn't resolve to either a mapping or string?

Reply:
Updated. Let me know what you think.